modules/configuration/src/main/java/org/apache/ignite/configuration/ConfigurationRegistry.java

```diff
@@ -71,6 +80,39 @@ public void registerStorage(ConfigurationStorage configurationStorage) {
         return (T)configs.get(rootKey.key());
     }
 
+    /**
+     * Convert configuration subtree into a user-defined representation.
+     *
+     * @param path Path to configuration subtree. Can be empty, can't be {@code null}.
+     * @param visitor Visitor that will be applied to the subtree and build the representation.
+     * @param <T> Type of the representation.
+     * @return User-defined representation constructed by {@code visitor}.
+     * @throws IllegalArgumentException If {@code path} is not found in current configuration.
+     */
+    public <T> T represent(List<String> path, ConfigurationVisitor<T> visitor) throws IllegalArgumentException {
+        SuperRoot mergedSuperRoot = changer.mergedSuperRoot();
+
+        Object node;
+        try {
+            node = ConfigurationUtil.find(path, mergedSuperRoot);
+        }
+        catch (KeyNotFoundException e) {
+            throw new IllegalArgumentException(e.getMessage());
+        }
+
+        if (node instanceof TraversableTreeNode)
+            return ((TraversableTreeNode)node).accept(null, visitor);
+
+        assert node == null || node instanceof Serializable;
```

Comment:
how can we get a null node here? if it is not found, we'll throw an exception, what is the other scenario?

Reply:
1. Configuration might not be fully initialized
2. Non-existent named list element is represented as null now. Should I change it btw?